internal/controllers/topology/cluster/cluster_controller.go

```diff
@@ -221,6 +204,31 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl.Re
 func (r *Reconciler) reconcile(ctx context.Context, s *scope.Scope) (ctrl.Result, error) {
 	var err error
 
+	// Get ClusterClass.
+	clusterClass := &clusterv1.ClusterClass{}
+	key := client.ObjectKey{Name: s.Current.Cluster.Spec.Topology.Class, Namespace: s.Current.Cluster.Namespace}
+	if err := r.Client.Get(ctx, key, clusterClass); err != nil {
+		return ctrl.Result{}, errors.Wrapf(err, "failed to retrieve ClusterClass %s", s.Current.Cluster.Spec.Topology.Class)
```

Comment:
Q: should we surface this use case in the topology reconcile condition given that we are accepting clusters without a corresponding ClusterClass?
This will make it easier for users to detect cases where e.g. they made a typo in the ClusterClass name

Reply:
Same question. Are we not surfacing all returned errors in the condition? Or do you mean with a special reason